src/mon/FSCommands.cc

```diff
@@ -255,12 +239,15 @@ class FsNewHandler : public FileSystemCommandHandler
     pg_pool_t const *metadata_pool = mon->osdmon()->osdmap.get_pg_pool(metadata);
     ceph_assert(metadata_pool != NULL);  // Checked it existed above
 
+    bool allow_overlay = false;
+    cmd_getval(cmdmap, "allow_dangerous_metadata_overlay", allow_overlay);
+
```

Comment:
Move it to above the _check_pool call for POOL_METADATA. Otherwise, LGTM.

Reply:
Done